File: Connectivity.java
import java.util.ArrayList;
import java.util.List;

public class Graph {
    private int V;
    private List<List<Integer>> adjList;

    public Graph(int V) {
        this.V = V;
        adjList = new ArrayList<>(V);
        for (int i = 0; i < V; i++) {
            adjList.add(new ArrayList<>());
        }
    }

    public void addEdge(int u, int v) {
        adjList.get(u).add(v);
        adjList.get(v).add(u); // For an undirected graph
    }

    public void DFS(int v, boolean[] visited) {
        visited[v] = true;
        for (int u : adjList.get(v)) {
            if (!visited[u]) {
                DFS(u, visited);
            }
        }
    }

    public boolean isConnected() {
        boolean[] visited = new boolean[V];
        int startVertex = 0; // Starting vertex for DFS
        DFS(startVertex, visited);

        for (boolean visit : visited) {
            if (!visit) {
                return false; // If there's an unvisited vertex, the graph is not connected
            }
        }

        return true; // If all vertices are visited, the graph is connected
    }
}

File: Connectivity.py
class Graph:
    def __init__(self, V):
        self.V = V
        self.adjList = [[] for _ in range(V)]

    def addEdge(self, u, v):
        self.adjList[u].append(v)
        self.adjList[v].append(u)  # For an undirected graph

    def DFS(self, v, visited):
        visited[v] = True
        for u in self.adjList[v]:
            if not visited[u]:
                self.DFS(u, visited)

    def isConnected(self):
        visited = [False] * self.V
        startVertex = 0  # Starting vertex for DFS
        self.DFS(startVertex, visited)

        return all(visited)
